subprojects/build-option/src/test/groovy/org/gradle/internal/buildoption/StringBuildOptionTest.groovy

```diff
@@ -116,6 +116,28 @@ class StringBuildOptionTest extends Specification {
         assertDeprecatedDescription(shortOption, deprecationWarning)
     }
 
+    def "can configure deprecated property option"() {
```

Comment:
:x: We should have a similar test case for all the other `BuildOption` types e.g. `ListBuildOptionTest` etc.